### PR TITLE
Input region tests

### PR DESCRIPTION
**Pull request by wmww**
Tests surfaces with input regions with the same test style as `test_surface_events.cpp` uses

### CODE REVIEW

File: tests/surface_input_regions.cpp [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+using namespace testing;
+
+// No tests use SUBTRACT_RECT as that is not yet implemented in Mir
+
+enum RegionAction
+{
+    ADD_RECT,
+    SUBTRACT_RECT
+};
```
Comment (RAOF) [inline tests/surface_input_regions.cpp]: We'd normally use an `enum class` and CamelCase for the elements

File: tests/surface_input_regions.cpp [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    {
+        switch(e.action)
+        {
+        case ADD_RECT:
+            wl_region_add(wl_region, e.x, e.y, e.width, e.height);
+            break;
+        case SUBTRACT_RECT:
+            wl_region_subtract(wl_region, e.x, e.y, e.width, e.height);
+            break;
+        default:
```
Comment (RAOF) [inline tests/surface_input_regions.cpp]: If you want to you can `#ifndef __clang__` here; clang will believe that the above is exhaustive

File: tests/surface_input_regions.cpp (hunk 1 above)
Comment (AlanGriffiths) [reply tests/surface_input_regions.cpp]: > We'd normally use an enum class and CamelCase for the elements

Actually our [guidelines](https://mir-server.io/doc/cppguide/index.html?showone=Enumerator_Names#Enumerator_Names) say lower_case for the members.

File: tests/surface_input_regions.cpp [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+
+using namespace testing;
+
+// No tests use Subtract as that is not yet implemented in Mir
+
+enum class RegionAction
+{
+    Add,
+    Subtract
+};
```
Comment (AlanGriffiths) [inline tests/surface_input_regions.cpp]: Sorry, I didn't realize I was commenting on an obsolete review.

The enum class members [should be](https://mir-server.io/doc/cppguide/index.html?showone=Enumerator_Names#Enumerator_Names) lower_case.

**Pull request merged**